src/business-logic/services/device-changed-service.ts

```diff
@@ -0,0 +1,102 @@
+import { DatabaseChangeService } from './interfaces/database-change-service'
+import { StatusMessageEventEmitter } from './interfaces/status-message-event-emitter'
+import { DataChangedListener } from '../../data-access/repositories/interfaces/data-changed-listener'
+import { Device } from '../../model/entities/device'
+import { StatusMessage } from '../../model/entities/status-message'
+import { StatusMessageRepository } from '../../data-access/repositories/interfaces/status-message-repository'
+import { NotFoundException } from '../../model/exceptions/not-found-exception'
+import { StatusCode } from '../../model/enums/status-code'
+
+export class DeviceChangedService implements DatabaseChangeService {
+  private static instance: DatabaseChangeService
+
+  public static getInstance(
+    statusMessageEventEmitter: StatusMessageEventEmitter,
+    statusMessageRepository: StatusMessageRepository,
+    deviceChangedListener: DataChangedListener<Device>
+  ): DatabaseChangeService {
+    if (!this.instance) {
+      this.instance = new DeviceChangedService(
+        statusMessageEventEmitter,
+        statusMessageRepository,
+        deviceChangedListener
+      )
+    }
+    return this.instance
+  }
+
+  constructor(
+    private readonly statusMessageEventEmitter: StatusMessageEventEmitter,
+    private readonly statusMessageRepository: StatusMessageRepository,
+    deviceChangedListener: DataChangedListener<Device>
+  ) {
+    this.listenForStatusMessageChanges(deviceChangedListener)
+  }
+
+  private listenForStatusMessageChanges(deviceChangedListener: DataChangedListener<Device>): void {
+    deviceChangedListener.onUpdated(device => void this.onDeviceUpdated(device))
+  }
+
+  private async onDeviceUpdated(device: Device): Promise<void> {
```

Comment:
At a glance, this method seems rather large and full of nesting logic. But i tried tampering a bit myself and i see why it is this way. So i don't have any suggestions to improve either :-D So i'm only commenting to make a note in case you had any ideas yourself, otherwise it doesn't change my overall opinion of the PR.